.github/workflows/docker-build-static.yml

```
@@ -453,14 +463,18 @@ jobs:
           # list of Docker images to use as base name for tags
           images: |
             nitnelave/lldap
+            ghcr.io/lldap/lldap
+          flavor:
+            suffix= ${{ matrix.suffix }}
           # generate Docker tags based on the following events/attributes
           tags: |
-            type=ref,event=branch
             type=ref,event=pr
-            type=semver,pattern={{version}}
-            type=semver,pattern={{major}}.{{minor}}
-            type=semver,pattern={{major}}
-            type=sha
+            type=semver,pattern={{raw}}
+            type=semver,pattern=v{{major}}
+            type=semver,pattern=v{{major}}.{{minor}}
+            type=semver,pattern=v{{major}}.{{minor}}.{{patch}}
+            type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', 'main') }}
+            type=raw,value=stable,enable=${{ startsWith(github.ref, 'refs/tags/v') }}
```

```suggestion
            type=semver,pattern=v{{version}}
            type=semver,pattern=v{{major}}.{{minor}}
            type=semver,pattern=v{{major}}
            type=semver,pattern={{version}},value=stable
            type=raw,value=latest,enable={{is_default_branch}}
```
More consistent way of handling latest and [stable](https://github.com/marketplace/actions/docker-metadata-action#typesemver) IMO - stable could also be converted to raw and use `enable=${{ github.event_name == 'release' }}`
Also, i don't think we need teh raw symver, we can add `v` to the parsed version

---

Already tried this `type=semver,pattern={{version}},value=stable`, the `value` must conform semver `vX.Y.Z`

This maybe useful,  `type=raw,value=latest,enable={{is_default_branch}}` checking which one is default branch, in this case `main` or future change it will follow.

```
            type=semver,pattern=v{{version}}
            type=semver,pattern=v{{major}}
            type=semver,pattern=v{{major}}.{{minor}}
            type=semver,pattern=v{{major}}.{{minor}}.{{patch}}
            type=raw,value=latest,enable={{ is_default_branch }}
            type=raw,value=stable,enable=${{ startsWith(github.ref, 'refs/tags/v') }}
```
This should be sufficient?

---

EDIT: please change ` type=semver,pattern={{version}},value=stable` to the propsed alternative - i misunderstood how value was going to work in that context

---

your alternative with the ref also works -roll with either

---

```
type=ref,event=pr
type=semver,pattern=v{{version}}
type=semver,pattern=v{{major}}
type=semver,pattern=v{{major}}.{{minor}}
type=raw,value=latest,enable={{ is_default_branch }},suffix=${{ matrix.noflavour }}
type=raw,value=stable,enable=${{ startsWith(github.ref, 'refs/tags/v') }},suffix=${{ matrix.noflavour }}
```

So basically on `latest` and `stable` override the suffix, become `:latest` `:latest-debian` and `:stable` `:stable-debian`